xtractmime/__init__.py

```diff
@@ -48,8 +69,33 @@ def extract_mime(
     extra_types: Optional[Tuple[Tuple[bytes, bytes, Set[bytes], str], ...]] = None,
 ) -> str:
     extra_types = extra_types or tuple()
-    # supplied_type = content_types[-1] if content_types else None
+    supplied_type = content_types[-1] if content_types else None
+    check_for_apache = http_origin and supplied_type in _APACHE_TYPES
+    resource_header = memoryview(body) if len(body) < 1445 else memoryview(body)[:1445]
+
+    if supplied_type in (None, "unknown/unknown", "application/unknown", "*/*"):
+        _find_unknown_mimetype(body, not no_sniff)
```

Comment:
So, I think we can pass `EXTRA_TYPES` (containing the table mentioned in step 2  of section 7.1) as a default parameter for `extra_types` instead of `None` into `extract_mime` which users can extend accordingly.

Reply:
That would allow users to alter the contents of that table, which the algorithm does not seem to allow. If I got it right, the algorithm allows to define additional rows _after_ the rows of that table. So users should only be able to add rows, not to change them.